druid-shell/src/platform/x11/application.rs

```diff
@@ -525,36 +549,57 @@ fn poll_with_timeout(conn: &Rc<XCBConnection>, idle: RawFd, timeout: Instant) ->
 
     // We start with no timeout in the poll call. If we get something from the idle handler, we'll
     // start setting one.
-    let mut poll_timeout = -1;
+    let mut honor_idle_timeout = false;
     loop {
         fn readable(p: PollFd) -> bool {
             p.revents()
                 .unwrap_or_else(PollFlags::empty)
                 .contains(PollFlags::POLLIN)
         };
 
+        // Compute the deadline for when poll() has to wakeup
+        let deadline = if honor_idle_timeout {
+            Some(earliest_timeout)
+        } else {
+            timer_timeout
+        };
+        // ...and convert the deadline into an argument for poll()
+        let poll_timeout = if let Some(deadline) = deadline {
+            if deadline < now {
+                0
+            } else {
+                c_int::try_from(deadline.duration_since(now).as_millis())
```

Comment:
I guess it would be best if this would round up instead of round down (in `as_millis()`). Since that does not seem to be easily possible: How about adding a `+1` to the result? That way, `poll` should sleep until "something interesting" happens. Otherwise, it is possible that `Instant::now()` is less than 1 ms before "something interesting" and the code here would busy loop with `poll()` with a timeout of zero...

Reply:
I think adding 1 ms is a good approach.